olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -235,8 +256,18 @@ OpenResult DiskCache::Open(const std::string& data_path,
     error_ = NoError{};
   }
 
+  if (is_read_only) {
+    const auto num_files_at_level0 = GetFilesNumberAtLevel0(*db);
```

Comment:
Same here. Call ContinueCompaction() method. It is not really important for the user to know how many L0 files there are, he only needs to know that compaction is not finished and he should compact the db before handing it over to us.